### PR TITLE
Fix LGTM errors in 3p code

### PR DESCRIPTION
**Pull request by rsimha-amp**
Fixes the following LGTM errors:

- [src/3p-frame.js#L302](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/a4d5dedaa4349a630d64ac4ffa41ed08b90f7efe/files/src/3p-frame.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L302)
- [3p/3p.js#L152](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/a4d5dedaa4349a630d64ac4ffa41ed08b90f7efe/files/3p/3p.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L152)

Suppresses this LGTM error:
- [3p/environment.js#L196](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/a4d5dedaa4349a630d64ac4ffa41ed08b90f7efe/files/3p/environment.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L196)

Partial fix for #12597

### CODE REVIEW

**Comment by rsimha-amp** [comment]
/to @cramforce @jridgewell 

@cramforce I assume the `(0,` in [3p/environment.js#L196](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/a4d5dedaa4349a630d64ac4ffa41ed08b90f7efe/files/3p/environment.js?sort=name&dir=ASC&mode=heatmap&excluded=false#L196) is intentional, so I left it in. We can silence that LGTM error in future checks.

File: 3p/3p.js [hunk 1]
```diff
@@ -149,7 +149,7 @@ export function validateSrcPrefix(prefix, src) {
     prefix = [prefix];
   }
   if (src !== undefined) {
-    for (let p = 0; p <= prefix.length; p++) {
```
Comment (rsimha-amp) [inline 3p/3p.js]: https://github.com/ampproject/amphtml/pull/2022/files#diff-b53dea7f09202bac0ac9b1ece2054d5aR116

**Comment by rsimha-amp** [comment]
Errors fixed: https://lgtm.com/projects/g/ampproject/amphtml/rev/pr-e34770f47d9c088e76ad011cf2926a4e5c8a467e

**Comment by jridgewell** [comment]
> I assume the (0, in 3p/environment.js#L196 is intentional, so I left it in. We can silence that LGTM error in future checks.

Correct. This causes an indirect function call, which has different semantics than a direct method call.

The most common pattern for indirect calls is `(0, fn)`. Can we allow only that?

**Comment by rsimha-amp** [comment]
@jridgewell I've suppressed the error.

**Comment by rsimha-amp** [comment]
Errors fixed / suppressed: https://lgtm.com/projects/g/ampproject/amphtml/rev/pr-65334aea1e4eccceddfbe957df9a6438c03d0bf8

**Pull request merged**